src/main/java/com/vishva/Payapp/service/impl/CreditServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package com.vishva.Payapp.service.impl;
+
+import com.vishva.Payapp.exchange.CreditRequest;
+import com.vishva.Payapp.exchange.CreditResponse;
+import com.vishva.Payapp.model.AccountEntity;
+import com.vishva.Payapp.repository.AccountRepository;
+import com.vishva.Payapp.repositoryservice.CreditRepositoryService;
+import com.vishva.Payapp.service.CreditService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import java.math.BigDecimal;
+import java.util.Optional;
+
+@Service
+public class CreditServiceImpl implements CreditService {
+
+    @Autowired
+    private CreditRepositoryService creditRepositoryService;
+
+    @Autowired
+    private AccountRepository accountRepository;
+
+    @Override
+    public CreditResponse amountCredited(CreditRequest creditRequest) {
+
+         CreditResponse creditResponse = new CreditResponse();
+
+         Optional<AccountEntity> optionalAccountEntity = creditRepositoryService.getAccountEntities(creditRequest.getAccId());
+         if(optionalAccountEntity.isEmpty()) {
+             creditResponse.setCreditStatus("Failure: No such account is present");
```

Comment:
I don't think is what i meant when i told you to return response specific to use case. Since we return http status at controller level, we should also prepare our response at controller level. That's why told you to read about controller advice. 

Read below article how we can handle exceptions at global level
http://zetcode.com/springboot/controlleradvice/

Reply:
Okay, I will read the given article.
Is it incorrect or just not what you were expecting. Or do you want me to give the same message but for the case but in the controller layer